src/components/SortableTable.tsx

```diff
@@ -0,0 +1,346 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components/macro';
+
+import { ArrowDownIcon } from 'icons';
+import { breakpoints, fonts, fontSizes } from 'styles';
+
+import LoadingBar from 'components/LoadingBar';
+
+export enum SortType {
+  String = 'String',
+  Number = 'Number',
+}
+
+export enum SortDirection {
+  Increasing = 'Increasing',
+  Decreasing = 'Decreasing',
+}
+
+export enum CellAlign {
+  Center = 'Center',
+  End = 'End',
+  Start = 'Start',
+}
+
+type SortByColumnConfig = {
+  key: string;
+  sortDirection: SortDirection;
+};
+
+type RowData = {
+  // eslint-disable-next-line
+  [key: string]: any;
+};
+
+type RenderHeaderCellProps = {
+  isSorted: boolean;
+  onClick?: () => void;
+} & BasicCellProps;
+
+type RenderCellProps = {
+  align?: CellAlign;
+  rowData: RowData;
+  rowIndex: number;
+};
+
+type RenderCellFunction = (props: RenderCellProps) => React.ReactNode;
+type RenderHeaderCellFunction = (props: RenderHeaderCellProps) => React.ReactNode;
+
+type BasicCellProps = {
+  align?: CellAlign;
+  fillWidth?: boolean;
+  isSortable?: boolean;
+  key: string;
+  label: React.ReactNode;
+  sortDirection?: SortDirection;
+};
+
+type ColumnConfig = {
+  renderCell: RenderCellFunction;
+  renderHeaderCell?: RenderHeaderCellFunction;
+  sort?: () => number;
+  sortType?: SortType;
+} & BasicCellProps;
+
+type SortableTableProps = {
+  columns: ColumnConfig[];
+  data: RowData[];
```

Comment:
Ideally `RowData` would be a "generic type variable" passed into each `<SortableTable>` instance or inferred somehow

Reply:
Oooo I really like the idea of passing it in as a generic, let's add that as an additional task independently of this PR tho.